blue_ai/envs/transient_goals.py

```diff
@@ -52,88 +57,114 @@ def __init__(
         self.transient_penalty = transient_penalty
         self.n_transient_obstacles = n_transient_obstacles
         self.transient_obstacles = transient_obstacles
-        self.replace_transient_obstacles = replace_transient_goals
+        self.replace_transient_obstacles = replace_transient_obstacles
 
         mission_space = MissionSpace(mission_func=self._gen_mission)
 
-        max_steps = 4 * width**2
+        ## This seems really arbitrary
+        max_steps = 4 * (len(self.im[0])) ** 2
 
         super().__init__(
             mission_space=mission_space,
-            width=width,
-            height=height,
+            width=len(self.im[0]),
+            height=len(self.im),
             # Set this to True for maximum speed
-            see_through_walls=True,
+            see_through_walls=False,
             max_steps=max_steps,
             agent_view_size=5,
             **kwargs,
         )
         self.actions = Actions
 
-    def step(self, action):
-        self.step_count += 1
+    def _replace_transient_obstacles(self, reward):
+        self.penalties.append(ObstacleNoTerminate(reward=reward))
+        self.place_obj(self.penalties[-1], max_tries=100)
 
-        mandatory = 0
-        optional = 0
-        reward = 0
-        terminated = False
-        truncated = False
+    """
+    Turns the agent to the left
+    """
 
-        # Get the position in front of the agent
-        fwd_pos = self.front_pos
+    def _turn_left(self):
+        self.agent_dir = (self.agent_dir - 1) % 4
+
+    """
+    Turns the agent to the right
+    """
 
-        # Get the contents of the cell in front of the agent
+    def _turn_right(self):
+        self.agent_dir = (self.agent_dir + 1) % 4
+
+    """
+    Determines if the current state is a end state
+    """
+
+    def _check_termination_conditions(self, cell_type: str):
+        return cell_type == "goal" or cell_type == "lava"
+
+    def _get_reward_amount(
+        self, fwd_cell: GoalNoTerminate | ObstacleNoTerminate | WorldObj
+    ):
+        kind = fwd_cell.type
+        reward = 0.0
+
+        if hasattr(fwd_cell, "reward"):
+            reward = fwd_cell.reward
+
+        if kind == "goal":
+            reward = self.termination_reward
+
+        return reward
+
+    def _handle_forward(self):
+        fwd_pos = self.front_pos
         fwd_cell = self.grid.get(*fwd_pos)
 
-        # Rotate left
-        if action == self.actions.left:
-            self.agent_dir -= 1
```

Comment:
We are now using the same logic for turning left and right. I love Python's implementation of Modulo; I know it differs from almost every other language, but this one is correct.